test/testdata/rbi/array.rb

```diff
@@ -48,3 +48,7 @@
 # errors
 
 T::Array[Float].new.sum.nan? # error: Method `nan?` does not exist on `Integer` component of `T.any(Integer, Float)`
+
+# bsearch
+T.reveal_type([1,2].bsearch {|x| x == 1}) # error: Revealed type: `T.nilable(Integer)`
+T.reveal_type([1,2].bsearch {|x| x}) # error: Revealed type: `T.nilable(Integer)`
```

Comment:
Could you add a test for the `bsearch_index` method that operates over an array of non-`Integer` values? It would also be great to check that searching arrays of non-`Integer` values with `bsearch` also produces the expected types.

Reply:
Done! Is that what you had in mind?